openml/extensions/sklearn/extension.py

```diff
@@ -1573,6 +1577,16 @@ def _prediction_to_probabilities(y: np.ndarray, classes: List[Any]) -> np.ndarra
             else:
                 model_classes = used_estimator.classes_
 
+            if not isinstance(model_classes, list):
+                model_classes = model_classes.tolist()
+
+            # to handle the case when dataset is numpy and categories are encoded
+            # however the class labels stored in task are still categories
+            if isinstance(y_train, np.ndarray) and \
+                    isinstance(typing.cast(List[str], task.class_labels)[0], str):
+                model_classes = [typing.cast(List[str], task.class_labels)[i]
```

Comment:
Do we actually have a unit test for this one?

Reply:
No, don't think so. Should I add a unit test in this PR or add a new PR?